omp/components/device_matrix_data_kernels.cpp

```diff
@@ -96,9 +104,13 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
 
 template <typename ValueType, typename IndexType>
 void sort_row_major(std::shared_ptr<const DefaultExecutor> exec,
-                    Array<matrix_data_entry<ValueType, IndexType>>& data)
+                    device_matrix_data<ValueType, IndexType>& data)
 {
-    std::sort(data.get_data(), data.get_data() + data.get_num_elems());
+    Array<matrix_data_entry<ValueType, IndexType>> tmp{exec,
+                                                       data.get_num_elems()};
+    soa_to_aos(exec, data, tmp);
+    std::sort(tmp.get_data(), tmp.get_data() + tmp.get_num_elems());
+    aos_to_soa(exec, tmp, data);
```

Comment:
Its unfortunate that the stdlib does not have a zip_iterator

Reply:
yes, we have IteratorFactory and the whole shebang, but it only works with pairs, not arbitrary tuples. Also a parallel std::sort would be really helpful

Reply:
#966 (: